src/robotFactory.js

```diff
@@ -1,10 +1,83 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
```

Comment:
and this way you can have assign the existing coords here
```suggestion
    this.coords = coords;
```